### PR TITLE
[TIMOB-4979] Android: Expose dragstart and dragend on ScrollView

### PR DESCRIPTION
**Pull request by m1ga**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-4979

```javascript
var win = Ti.UI.createWindow({
	backgroundColor: "#fff"
});
var scrollView = Ti.UI.createScrollView({
	height: Ti.UI.FILL,
	width: Ti.UI.FILL,
	contentHeight: 4000,
	scrollType: "vertical"
})

function onScrolling(e) {
	console.log("Scrolling");
}

function onDragstart(e) {
	console.log("---Drag start---");
}

function onDragend(e) {
	console.log("---Drag end---");
}
scrollView.addEventListener("scroll", onScrolling);
scrollView.addEventListener("dragstart", onDragstart);
scrollView.addEventListener("dragend", onDragend);
win.add(scrollView);
win.open();
```

### CODE REVIEW

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIScrollView.java [hunk 1]
```diff
@@ -365,6 +394,12 @@ protected void onScrollChanged(int l, int t, int oldl, int oldt)
 		{
 			super.onScrollChanged(l, t, oldl, oldt);
```
Comment (garymathews) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIScrollView.java]:
```Java
KrollDict data = new KrollDict();
```

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIScrollView.java [hunk 2]
```diff
@@ -249,6 +262,12 @@ protected void onScrollChanged(int l, int t, int oldl, int oldt)
...
```
Comment (garymathews) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIScrollView.java]:
```Java
KrollDict data = new KrollDict();
```

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIScrollView.java [hunk 3]
```diff
@@ -249,6 +262,12 @@ protected void onScrollChanged(int l, int t, int oldl, int oldt)
...
 
+			if (!isScrolling && isTouching) {
+				isScrolling = true;
+				data = new KrollDict();
+				getProxy().fireEvent(TiC.EVENT_DRAGSTART, data);
+			}
+
 			KrollDict data = new KrollDict();
```
Comment (garymathews) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIScrollView.java]:
```Java
data = new KrollDict();
```

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIScrollView.java [hunk 4]
```diff
@@ -249,6 +262,12 @@ protected void onScrollChanged(int l, int t, int oldl, int oldt)
 		{
 			super.onScrollChanged(l, t, oldl, oldt);
 
+			if (!isScrolling && isTouching) {
+				isScrolling = true;
+				data = new KrollDict();
```
Comment (garymathews) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIScrollView.java]: Remove this line

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIScrollView.java [hunk 5]
```diff
@@ -312,6 +331,16 @@ public TiScrollViewLayout getLayout()
 
 		@Override
 		public boolean onTouchEvent(MotionEvent event) {
+			if (event.getAction() == MotionEvent.ACTION_MOVE && !isTouching) {
```
Comment (garymathews) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIScrollView.java]: Should probably put these conditions below:
```Java
if (event.getAction() == MotionEvent.ACTION_MOVE && !mScrollingEnabled) {
```
to not fire when `scrollingEnabled` is false?

**Review by garymathews — CHANGES_REQUESTED** [review]

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIScrollView.java (hunk 5 above)
Comment (m1ga) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIScrollView.java]: sure, makes sense!

**Review by garymathews — APPROVED** [review]

**Comment by hansemannn** [comment]
@garymathews Can we move on to QE-testing?

**Comment by garymathews** [comment]
@hansemannn Yes 👍

**Review by ssjsamir — APPROVED** [review]
FR Passed - was able to see `dragstart` and `dragend` on the scroll view from the console. 

**Test Steps**
+ Copied the test case from the description above
+ Tested the test case with the build from this PR
+ Ran the program on android emulator (Nexus 6 running Android 7.0)
+ Scrolled up and down 
+ `dragstart` and `dragend`  was viewable in the console 

**Output from the console:**
```
[INFO] :   ---Drag start---
[INFO] :   Scrolling
[INFO] :   Scrolling
[INFO] :   Scrolling
[INFO] :   Scrolling
[INFO] :   Scrolling
[INFO] :   Scrolling
[INFO] :   Scrolling
[INFO] :   Scrolling
[INFO] :   ---Drag end---
[INFO] :   Scrolling
[INFO] :   Scrolling
[INFO] :   Scrolling
```
**Test environment**
Appcelerator Command-Line Interface, version 6.2.0
Android Emulator (Nexus 6 running android 7.0)
Operating System Name: Mac OS X El Capitan
Operating System Version: 10.11.6
Node.js Version: 4.6.0
npm: 4.2.9
Xcode: 8.2
Appcelerator Studio: 4.8.1.201612050850

**Pull request merged**